apps/web/src/app/wallet/hooks/use-nfts.ts

```diff
@@ -0,0 +1,91 @@
+import { useNavigate } from '@tanstack/react-router'
+import { useCallback, useEffect } from 'react'
+
+import { useToast } from 'src/app/core/hooks/use-toast'
+import { Toast } from 'src/app/core/services/toast'
+import { modalService } from 'src/components/organisms/modal/provider'
+import { a } from 'src/interfaces/cms/useAssets'
+import { c } from 'src/interfaces/cms/useContent'
+import { queryClient } from 'src/interfaces/query-client'
+
+import { WalletQueryKeys } from '../queries/query-keys'
+import { useClaimNft } from '../queries/use-claim-nft'
+import { WalletPagesPath } from '../routes/types'
+import { Nft } from '../services/wallet/types'
+
+export const useNfts = () => {
+  const toast = useToast()
+  const navigate = useNavigate()
+
+  const nftModalKey = 'nft'
+
+  const exit = useCallback(() => {
+    navigate({
+      to: WalletPagesPath.NFTS,
+      search: undefined,
+      replace: true,
+    })
```

Comment:
```suggestion
      router.navigate({
        search: undefined,
        replace: true,
      })
    navigate({
      to: WalletPagesPath.NFTS,
    })
```

Wdyt about this? With that, we can keep the previous route, but also reset the "search"

Reply:
I had some unexpected behavior when using router.navigate, and with this approach both the redirect and the header handleBack worked properly. Do you see any extra complications with this?